From 909d77066640baf07e01932b6aa5c719c50f1ccd Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 12 Jun 2022 08:28:20 -0400
Subject: [PATCH] eliminate lint warning

---
 coders/json.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/json.c b/coders/json.c
index c461689fd..c2bbb6153 100644
--- a/coders/json.c
+++ b/coders/json.c
@@ -323,6 +323,8 @@ static void ColorFormatLocaleFile(FILE *file,const char *format,Image *image,
   MagickPixelPacket
     pixel;
 
+  assert(p != (const PixelPacket *) NULL);
+  assert(index != (const IndexPacket *) NULL);
   GetMagickPixelPacket(image,&pixel);
   SetMagickPixelPacket(image,p,index,&pixel);
   GetColorTuple(&pixel,MagickTrue,color);
